src/components/GoodsList.js

```diff
@@ -0,0 +1,117 @@
+import React from 'react';
+
+import PropTypes from 'prop-types';
+
+export class GoodsList extends React.Component {
+  state = {
+    isReversed: false,
+    sortBy: '',
+    length: 1,
+  }
+
+  reverse = () => {
+    this.setState(state => ({
+      isReversed: !state.isReversed,
+    }));
+  }
+
+  sortByLength = () => {
+    this.setState({ sortBy: 'length' });
+  }
+
+  sortByAlphabet = () => {
+    this.setState({ sortBy: 'alphabet' });
+  }
+
+  reset = () => {
+    this.setState({
+      isReversed: false,
+      sortBy: '',
+      length: 1,
+    });
+  }
+
+  selectChange = (value) => {
+    this.setState({ length: value });
+  }
+
+  render() {
+    const { isReversed, sortBy, length } = this.state;
+    const selectArr = Array(10).fill(0).map((_, i) => i + 1);
```

Comment:
is better create array outside the render method, at that moment you create a new array every render